pom.xml

```diff
@@ -403,7 +403,7 @@
           <configuration>
             <argLine>-Xms512m -Xmx3g -Ddrill.exec.http.enabled=false
               -Ddrill.exec.sys.store.provider.local.write=false
-              -Dorg.apache.drill.exec.server.Drillbit.system_options="org.apache.drill.exec.compile.ClassTransformer.scalar_replacement=on"
+              -Dorg.apache.drill.exec.server.Drillbit.system_options="org.apache.drill.exec.compile.ClassTransformer.scalar_replacement=try"
```

Comment:
Let me expound. We have Drill setup so that tests execute with scalar replacement fails the test if it fails. This is to ensure that we don't accidentally break the scalar replacement functionality. There are a small number of situations where scalar replacement functionality doesn't work (certain expression patterns). For those tests, (and only those tests), we set scalar replacement to "try" manually as part of those tests. The try behavior is for scalar replacement to silently fail. 

I think there must be more dynamics on this bug. In your example here, you aren't actually changing the production behavior at all. The only change you made was to how we run tests. The change you made means that the tests are running the same as what people would expect in production. If your test above passes with "try" and you don't get an exception, that means that everything is working as it should be (e.g. no bug). However, Khurram is claiming that the query fails in production (try) mode. It sounds like you should work with him to understand whether his setup. It may be that it is accidentally in "on" instead of "try". If he is in "try" mode, then your test is not reproducing the problem.  Otherwise, this patch would fail to execute.